### ISSUE TITLE
Reference and Geo data types

### ISSUE THREAD

**Issue by jonmountjoy** [alickbass/CodableFirebase#13]
The [Firestore docs](https://firebase.google.com/docs/firestore/manage-data/data-types) mention Reference and Geographical Point data types.  Would it be possible to extend CodableFirebase to support these?  I use CodableFirebase for simple Struct types - but I've no idea how to start using the other data types. 

I believe the Geo point comes down as a [FirGeopoint](https://firebase.google.com/docs/firestore/reference/ios/firestore/api/reference/Classes#firgeopoint) . Which isn't Codable.  And the reference as a DocumentReference.

**Comment by alickbass** [comment]
@jonmountjoy thanks for reaching out! there are several things here that I want to point out. 

Firstly, so far this library has no external dependencies. This means, that you could potentially use it just for decoding your custom types into plain `Dictionary` without any notion of Firebase itself. Supporting these 2 types will mean that this library will have to have Firebase as a dependency. We can somehow avoid this by using protocols and asking people to add some code in their program if they want to use this type. This can easily solve the `FirGeopoint` support.

With `DocumentReference`, it's going to be much harder. In general, if your type conforms to `Codable`, this means that it could be potentially be used with any type of encoder/decoder, like `JSONEncoder`, `PlistEncoder` and it becomes really unclear how to encode and decode the DocumentReference type.

To sum up, I think, it's worth trying to support `FirGeopoint`, but I am not sure that it's a good idea to try to support `DocumentReference`

**Comment by alickbass** [comment]
But I think that your point is valid and I will try to play around with them and see what I can do

**Comment by alickbass** [comment]
@jonmountjoy I have created a PR that adds support for the `GeoPoint` and `DocumentReference`. README was also updated and now includes some instructions on how to add support for those types.  You are welcome to check it out :)

**Comment by jonmountjoy** [comment]
Thank you @alickbass !
